src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_INDEX = 1.5;
+    private Object[] elements;
```

Comment:
```suggestion
    private T[] elements;
```
Using T as the type parameter for the elements array leverages the class's type parameter, T, to define the type of objects that the collection will store. This change enhances type safety across the class, ensuring that all elements in the collection are of the same type as specified by T.

Reply:
changed